spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessorTests.java

```diff
@@ -273,6 +278,42 @@ public void queuesToDeclare() {
 		context.close();
 	}
 
+	@Test
+	public void concurrency() throws InterruptedException, ExecutionException {
+		final int concurrencyLevel = 8;
+		final ExecutorService executorService = Executors.newFixedThreadPool(concurrencyLevel);
+		try {
+			for (int i = 0; i < 1000; ++i) {
```

Comment:
@artembilan The race conditions was happening because of the contention of the first two requests. After that type information is cached. So it does not make a lot of sense to make 1000 iterations using cached data.